readme.md

```diff
@@ -1,57 +1,79 @@
-
-var appContext = mannish()
-```
+Originally inspired by Nicholas Zakas's talk on "[Creating A Scalable JavaScript Application Architecture](http://youtu.be/b5pFv9NB9fs)", this library fills the role of what he called the "sandbox" more closely than it serves the purpose he had for the mediator pattern.
 
-## Subscribe, adding yourself as a provider
+If you want a similar library to be used for broadcasting (something Mannish supported in older versions), I would recommend something like:
 
 ```js
-appContext.subscribe('unixTimestamp', function(someDate, cb) {
-	cb(null, Math.round(someDate.getTime() / 1000))
-})
+function makeBroadcaster() {
+	const emitter = new EventEmitter()
+
+	return {
+		emit(...args) {
+			emitter.emit(...args)
+		},
+		on(name, listener) {
+			emitter.on(name, listener)
+
+			return function removeListener() {
+				emitter.removeListener(name, listener)
+			}
+		}
+	}
+}
 ```
 
-Mannish uses promises internally to keep Zalgo at bay and prevent multiple subscribers from answering the same request.
+Mannish focuses on calling single functions and getting a response back.
+
+# API
+
+## `mediator = mannish()`
+
+This library exports a single function that returns a new mediator.
+
+<!-- js
+const mannish = require('./')
 
-If you just want to subscribe to events and it doesn't make sense to provide a response value, just ignore the callback.
+const mediator = mannish()
+-->
 
-If you want to use promises, then instead of calling the callback you can just return a promise instead.
+## `mediator.provide(name, function)`
 
-## Publish a request for something
+Supplies a function to handle all calls to the given name.
 
 ```js
-appContext.publish('unixTimestamp', new Date(), function(err, result) {
-	// result is a ten-digit number, wheee
+mediator.provide('pie', recipe => {
+	Array.isArray(recipe.contents) // => true
+
+	return 'I am a sweet pie'
 })
 ```
 
-The callback function is totally optional.  You can publish data objects to the app without any need for a response.
+## `promise = mediator.call(name, ...arguments)`
+
+Call whatever function is registered with the string `name`, with all other arguments being passed straight through.
+
+No matter what the other function returns, `call` returns a `Promise` containing the value it resolves to.
 
 ```js
-appContext.subscribe('flex', function(muscle) {
-	muscle.squeeze()
+const recipe = { type: 'recipe', contents: [ 'Step 1: find a real cook book' ] }
+mediator.call('pie', recipe).then(pie => {
+	pie // => 'I am a sweet pie'
 })
-
-appContext.publish('flex', new Muscle())
 ```
 
-## Remove all subscribers
+# Open questions
 
-```js
-appContext.removeAllListeners()
-```
+- Is it ever necessary to remove all providers?
+- Is it ever necessary to remove a single provider?
+	- If so, should anyone be able to remove the provider, or just the code that added it? (Should the mediator expose `removeProvider` or should `provide` return a removal function?)
```

Comment:
Maybe so, but I couldn't think of a real-life use case so I wanted to hold off until it was necessary.